commit 6ddd60127eb1a7c2050e89b5d9e0bf2569f877bc
Author: Tuetuopay <tuetuopay@me.com>
Date:   Mon Apr 9 10:34:31 2018 +0200

    cache: make "result" output argument for nl_cache_mngr_add() optional

    [thaller@redhat.com: rewritten commit message]

    https://github.com/thom311/libnl/issues/178
    https://github.com/thom311/libnl/pull/183

--- a/lib/cache_mngr.c
+++ b/lib/cache_mngr.c
@@ -415,6 +415,9 @@ int nl_cache_mngr_add_cache_v2(struct nl
  * the socket and call nl_cache_mngr_data_ready() to allow the library
  * to process netlink notification events.
  *
+ * @note Versions up to 3.4.0 actually required the result argument, preventing
+ * 	 NULL to be passed.
+ *
  * @see nl_cache_mngr_poll()
  * @see nl_cache_mngr_data_ready()
  *
@@ -445,7 +448,8 @@ int nl_cache_mngr_add(struct nl_cache_mn
 	if (err < 0)
 		goto errout_free_cache;
 
-	*result = cache;
+	if (result)
+		*result = cache;
 	return 0;
 
 errout_free_cache:
